client/app/hosting/offer/upgrade/hosting-offer-upgrade.html

```diff
@@ -1,114 +1,90 @@
-                            </select>
-                            <span class="oui-icon oui-icon-chevron-down" aria-hidden="true"></span>
-                        </div>
-                    </div>
-
-                    <div data-ng-if="ctrl.model.capacity && ctrl.model.capacity === 'KS'">
-                        <p data-translate="hosting_dashboard_service_offer_downgrade_KIMSUFI_2015"></p>
-                        <oui-checkbox id="downgradeAgree"
-                            data-model="ctrl.downgradeAgree"
-                            data-text="{{:: 'hosting_dashboard_service_offer_downgrade_KIMSUFI_2015_agree' | translate }}">
-                        </oui-checkbox>
-                    </div>
-                </form>
-
-                <div class="alert alert-info mt-5" role="alert"
-                     data-translate="hosting_order_upgrade_select_noduration">
-                </div>
-            </div>
-        </div>
-
-        <!-- Step 2: Durations -->
-        <div data-wizard-step
-             data-wizard-step-on-load="getDurations"
-             data-wizard-step-valid="ctrl.model.duration && ctrl.durations.available">
-
-            <p data-translate="hosting_order_upgrade_select_duration"
-               data-ng-if="ctrl.durations.available"></p>
-
-            <div class="text-center" data-ng-if="ctrl.loading.durations">
-                <oui-spinner></oui-spinner>
-                <div data-translate="hosting_order_upgrade_loading_order"></div>
-            </div>
-
-            <form name="orderUpgradeForm" data-ng-if="!ctrl.loading.durations">
-                <div class="oui-radio"
-                     data-ng-repeat="duration in (ctrl.durations.available | orderBy: 'duration': false) track by $index">
-                    <input type="radio" class="oui-radio__input" id="upgradeDuration" name="upgradeDuration"
-                           data-ng-model="ctrl.model.duration"
-                           data-ng-value="duration">
-                    <label class="oui-radio__label-container" for="upgradeDuration">
-                        <span class="oui-radio__label">
-                            <span data-ng-bind="duration.duration | wucDuration: 'longDate'"></span>&nbsp;:
-                            <span data-ng-bind-html="duration.prices | price: ctrl.ovhSubsidiary"></span>
-                        </span>
-                    </label>
+<div data-ng-controller="HostingUpgradeOfferCtrl as $ctrl">
+    <oui-modal data-heading="{{:: 'hosting_order_upgrade_modal_header' | translate }}"
+               data-on-dismiss="resetAction()"
+               data-loading="$ctrl.loading.availableOffers">
+        <oui-message
+          data-type="warning"
+          data-ng-if="isEmpty($ctrl.availableOffers)"
+          data-dismissable="false">
+          <span data-translate="hosting_order_upgrade_no_offer"></span>
+        </oui-message>
+
+        <oui-stepper data-ng-if="!isEmpty($ctrl.availableOffers)"
```

Comment:
Prefer using `<oui-stepper>` outside of `<oui-modal>` (if possible).

Reply:
Yes I'm gonna kill the modal and put that in a real page.

Reply:
After discussing that in weekly: We're going to prod that as-is and move it to a page in a further release.

Reply:
Could you please create a ticket which target theses improvements then?

Thanks!